DigitalLearningSolutions.Data.Tests/Services/GroupServiceTests/GroupsServiceAddCourseTests.cs

```diff
@@ -0,0 +1,597 @@
+﻿namespace DigitalLearningSolutions.Data.Tests.Services.GroupServiceTests
+{
+    using System;
+    using Castle.Core.Internal;
+    using DigitalLearningSolutions.Data.Models.Email;
+    using DigitalLearningSolutions.Data.Tests.TestHelpers;
+    using FakeItEasy;
+    using FluentAssertions.Execution;
+    using NUnit.Framework;
+
+    public partial class GroupsServiceTests
+    {
+        [Test]
+        public void AddCourseToGroup_adds_new_group_customisations_record()

```

Comment:
Because so much of this file is effectively copied and pasted tests from the GroupServiceEnrolDelegateTests, do you think we can be cheeky and get rid of most of them? If not, we are basically running the same tests twice. 

Now that we have multiple methods calling the Enrolling logic, and given it is somewhat substantial, might it be worth moving it into its own service, so we only need to test things once?

If keeping these tests, I've noticed there are a couple of tests in GroupServiceEnrolDelegateTests that don't seem to have an equivalent here. Namely:

- EnrolDelegateOnGroupCourses_update_uses_course_supervisor_id_if_course_supervisor_id_is_not_null
- EnrolDelegateOnGroupCourses_updates_existing_progress_record

Is there a specific reason those haven't been brought over?